jib-core/src/main/java/com/google/cloud/tools/jib/cache/Cache.java

```diff
@@ -130,6 +130,19 @@ public CachedLayer writeTarLayer(DescriptorDigest diffId, Blob compressedBlob)
     return cacheStorageWriter.writeTarLayer(diffId, compressedBlob);
   }
 
+  /**
+   * Writes a container configuration to {@code (cache directory)/local/config/(image id)}.
```

Comment:
I'd like to add something like

"An image ID is a SHA hash of a container configuration JSON. The value is also shown as IMAGE ID in `docker images`."

But as I commented in the design doc, this can still be confusing without a correct context. Basically, you can put anything for `imageId` to associate it with `containerConfiguration`, and almost 100% time, the actual SHA of the saved `containerConfiguration` will be different from `imageId`.

Reply:
To be clear, I don't think it really matters that the SHA of the saved configuration will be different from the image ID; what matters is that there's a consistent relationship from the SHA reported by the docker daemon image to the cached configuration. In other words, it's less "the configuration saved here is identical to the docker daemon image configuration with the same SHA" and more "we've seen a configuration with this SHA before, and we saved the configuration here after processing it a bit". But yeah, I can update some of the comments to make sure the misconception is less likely to happen.